ADAL/src/public/ADTelemetry.h

```diff
@@ -34,28 +34,9 @@
 
 /*!
     Callback function that will be called by ADAL when telemetry events are flushed.
-    @param  event        An event is represented by an array of ADTelemetryProperty.
+    @param  event        An event is represented by a dictionary of key-value properties.
  */
-- (void)dispatchEvent:(NSArray*)event;
-
-@end
-
-/*!
-    @class ADTelemetryProperty
- 
-    Each telemetry event dispatched to the above callback function is represented by an array of ADTelemetryProperty.
-    It includes property name and property value.
- */
-@interface ADTelemetryProperty : NSObject
-{
-    NSString* _name;
-    NSString* _value;
-}
-
-@property (readonly) NSString* name;
-@property (readonly) NSString* value;
-
-- (id)initWithName:(NSString*)name value:(NSString*)value;
+- (void)dispatchEvent:(nonnull NSDictionary<NSString*, NSString*> *)event;
```

Comment:
hmm ok...you'd still expect the delegate interface to be a little more structured around an "event" though. The dictionary interface is so open ended that the developer has no idea what they're going to get.

The parallel would be something like `ADUserInformation`. Yes, you could pass back all the information about a user with a dictionary (because that's all that we need), but it's nice to work with something typed.